tokio-compat/src/runtime/threadpool/mod.rs

```diff
@@ -0,0 +1,284 @@
+mod background;
+mod builder;
+pub use builder::Builder;
+
+use super::compat;
+use background::Background;
+
+use futures_01::future::Future as Future01;
+use futures_03_util::{compat::Future01CompatExt, future::FutureExt};
+use std::{future::Future, io};
+use tokio_executor::enter;
+use tokio_executor::threadpool::{Sender, ThreadPool};
+use tokio_net::driver;
+use tokio_timer::timer;
+
+pub struct Runtime {
+    inner: Option<Inner>,
+}
+
+#[derive(Debug)]
+struct Inner {
+    /// Task execution pool.
+    pool: ThreadPool,
+
+    /// Tracing dispatcher
+    trace: tracing_core::Dispatch,
+
+    /// Maintains a reactor and timer that are always running on a background
+    /// thread. This is to support `runtime.block_on` w/o requiring the future
+    /// to be `Send`.
+    ///
+    /// A dedicated background thread is required as the threadpool threads
+    /// might not be running. However, this is a temporary work around.
+    background: Background,
+
+    /// Compatibility background thread.
+    ///
+    /// This maintains a `tokio` 0.1 timer and reactor to support running
+    /// futures that use older tokio APIs.
+    compat_bg: compat::Background,
+}
+
+#[derive(Clone)]
+struct CompatSender<S>(S);
+
+// ===== impl Runtime =====
+
+/// Start the Tokio runtime using the supplied future to bootstrap execution.
```

Comment:
It'd be nice if the docs for each user-visible function were clear on what sort of future they took.